vtex-apps/glovo-integration
# Feat/settings update

This PR is an update to the [PR #29 - Glovo settings](https://github.com/vtex-apps/glovo-integration/pull/29).
Although it seems like a lot of changes, must of the changes are just an update in naming conventions for `types` to keep backend and frontend aligned.

To-do:
- Update `types` to be more clear and better structured.


#### What problem is this solving?
Features: 
- Add GraphQL and replace using `apps-graphql` with VTEX's native `apps` client to get and save the app settings.
- Add validation for the settings in the backend (`/node/resolvers/saveAppSettings.ts`). 

Fixes:
- Fix problems with recursion and validation in the `validateInputs` utility function.

Chores:
- Updated `types` in general.

#### How to test it?
The app is linked in [this workspace](https://glovoadmin--vtexspain.myvtex.com/admin/glovo-integration).
You should be able to save the settings as before but shouldn't be able to save them if somethings missing or incorrect.
